tests/integration/targets/aci_access_spine_switch_selector/tasks/main.yml

```diff
@@ -0,0 +1,162 @@
+# Test code for the ACI modules
+# Copyright: (c) 2017, Bruno Calogero <brunocalogero@hotmail.com>
+# Adapted from aci_switch_leaf_selector
+# Copyright: (c) 2023, Eric Girard <@netgirard>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact: 
+   aci_info: &aci_info
+    host: "{{ aci_hostname }}"
+    username: "{{ aci_username }}"
+    password: "{{ aci_password }}"
+    validate_certs: "{{ aci_validate_certs | default(false) }}"
+    use_ssl: "{{ aci_use_ssl | default(true) }}"
+    use_proxy: "{{ aci_use_proxy | default(true) }}"
+    output_level: "{{ aci_output_level | default('info') }}"
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: Deleting Switch Policy Spine profile exists for kick off
+    cisco.aci.aci_access_spine_switch_profile:
+      <<: *aci_info
+      spine_profile: sw_name_test
+      state: absent
+
+  - name: Ensuring Switch Policy Spine profile exists for kick off
+    cisco.aci.aci_access_spine_switch_profile: &aci_access_spine_switch_profile_present
+      <<: *aci_info
+      spine_profile: sw_name_test
+      state: present
+    register: spine_profile_present
+
+  # TODO: Ensure that spine Policy Group Exists (module missing) (infra:AccPortGrp)
```

Comment:
This module `aci_interface_policy_leaf_policy_group` says it manages `infra:AccPortGrp` although not sure if it works for spine group policies. However, `aci_rest` is always a possibility. 😸

Reply:
The comment is wrong as the class in question is infra:SpineAccNodePGrp not infra:AccPortGrp. I removed that comment and overall, modified all the old test cases to fit our current standards on integration test.